src/bin/tikv-server.rs

```diff
@@ -381,192 +382,207 @@ fn get_rocksdb_db_option(config: &toml::Value) -> RocksdbOptions {
 
-    let use_direct_io_for_flush_and_compaction = get_toml_boolean(config,
-                                                                  "rocksdb.use-direct-io-for-flush-and-compaction",
-                                                                  Some(false));
-    opts.set_use_direct_io_for_flush_and_compaction(use_direct_io_for_flush_and_compaction);
+    let direct_io = get_toml_boolean(config,
+                                     "rocksdb.use-direct-io-for-flush-and-compaction",
+                                     Some(false));
+    opts.set_use_direct_io_for_flush_and_compaction(direct_io);
 
     opts
 }
 
+struct CfOptValues {
+    pub block_size: i64,
+    pub block_cache_size: i64,
+    pub cache_index_and_filter_blocks: bool,
+    pub use_bloom_filter: bool,
+    pub whole_key_filtering: bool,
+    pub bloom_bits_per_key: i64,
+    pub block_based_filter: bool,
+    pub compression_per_level: String,
+    pub write_buffer_size: i64,
+    pub max_write_buffer_number: i64,
+    pub min_write_buffer_number_to_merge: i64,
+    pub max_bytes_for_level_base: i64,
+    pub target_file_size_base: i64,
+    pub level_zero_file_num_compaction_trigger: i64,
+    pub level_zero_slowdown_writes_trigger: i64,
+    pub level_zero_stop_writes_trigger: i64,
+}
+
+impl Default for CfOptValues {
+    fn default() -> CfOptValues {
+        CfOptValues {
+            block_size: 64 * KB as i64,
+            block_cache_size: 256 * MB as i64,
+            cache_index_and_filter_blocks: true,
+            use_bloom_filter: false,
+            whole_key_filtering: true,
+            bloom_bits_per_key: 10,
+            block_based_filter: false,
+            compression_per_level: String::from("no:no:lz4:lz4:lz4:zstd:zstd"),
+            write_buffer_size: 128 * MB as i64,
+            max_write_buffer_number: 5,
+            min_write_buffer_number_to_merge: 1,
+            max_bytes_for_level_base: 512 * MB as i64,
+            target_file_size_base: 32 * MB as i64,
+            level_zero_file_num_compaction_trigger: 4,
+            level_zero_slowdown_writes_trigger: 20,
+            level_zero_stop_writes_trigger: 36,
+        }
+    }
+}
+
 fn get_rocksdb_cf_option(config: &toml::Value,
                          cf: &str,
-                         block_cache_default: u64,
-                         use_bloom_filter: bool,
-                         whole_key_filtering: bool)
+                         default_values: CfOptValues)
                          -> RocksdbOptions {
     let prefix = String::from("rocksdb.") + cf + ".";
-    let mut opts = RocksdbOptions::new();
     let mut block_base_opts = BlockBasedOptions::new();
     let block_size = get_toml_int(config,
                                   (prefix.clone() + "block-size").as_str(),
-                                  Some(64 * 1024));
+                                  Some(default_values.block_size));
```

Comment:
I prefer to set `CfOptValues ` from config first, and implement a function for `CfOptValues ` to generate `RocksdbOptions` together.